internal/pkg/agent/application/upgrade/upgrade_test.go

```diff
@@ -983,3 +1000,60 @@ func Test_selectWatcherExecutable(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSameReleaseVersion(t *testing.T) {
+	tests := []struct {
+		name    string
+		current agentVersion
+		target  string
+		expect  bool
+	}{{
+		name: "current version is snapshot",
+		current: agentVersion{
+			version:  "1.2.3",
+			snapshot: true,
+		},
+		target: "1.2.3",
+		expect: false,
+	}, {
+		name: "target version is snapshot",
+		current: agentVersion{
+			version: "1.2.3",
+		},
+		target: "1.2.3-SNAPSHOT",
+		expect: false,
+	}, {
+		name: "target version is different version",
+		current: agentVersion{
+			version: "1.2.3",
+		},
+		target: "1.2.4",
+		expect: false,
+	}, {
+		name: "target version is same with pre-release",
```

Comment:
Maybe we should avoid "is same" in testcase names for a function that is called `isSame*` where the expected result is in fact that those versions are *not* the same
```suggestion
		name: "target version has same major.minor.patch, different pre-release",
```